src/memory_server.ml

```diff
@@ -278,22 +298,51 @@ let calculate_boot_time_host_free_memory () =
       pages_to_kib (Int64.of_nativeint boot_time_host_free_pages) in
     Int64.mul 1024L boot_time_host_free_kib
 
+let calc_constant_boot_time_host_free_memory constant_count interval =
+  debug "Check boot time host free memory: constant-count=%d check-interval=%f(seconds)" constant_count interval;
+  let rec calc_constant existing_results =
+    let free_memory = calculate_boot_time_host_free_memory () in
+    debug "Calculating boot time host free memory: %Ld" free_memory;
+    if (List.length existing_results) = (constant_count - 1) then
```

Comment:
If you want a constant-sized queue, where you always drop the oldest results you could allocate an array in the caller, and use it similar to a ring-buffer (i.e. you take index modulo array size, and you just keep incrementing the index).

Reply:
Yes, but in this case, we could refine the algorithm and only needs remeber the last free memory value.